tests/test_arithmetics.rs

```diff
@@ -505,6 +550,77 @@ mod return_not_implemented {
     #[pyclass]
     struct RichComparisonToSelf {}
 
+    #[cfg(Py_3_8)]
+    #[pymethods]
+    impl RichComparisonToSelf {
```

Comment:
To avoid passing this `#[cfg]` pain onto users, do you think we can make it so that users always write three-argument `__ipow__`, and then on Python 3.7 we just always fill in `None` as the third argument?

Reply:
I'm not sure how to implement this approach and not sure if it's correct. How will the user know he wont ever receive values in Py 3.7 (besides doc?) I think compile error and strict adjustment is better...?
The issue happens where the extract layer of PyO3 tries to convert Option<u32> when it's actually an invalid value... The interface would be then OptionInvalid<u32> where it will check for python version and do it based on that..?

Reply:
Agreed that compiler errors are more explicit.

I was thinking that given this is such an edge case anyway, and Python code using `=**` will always pass `None`, it's only if users explicitly invoke `__ipow__` that there's ever a non-none value. So it doesn't seem so bad if we say "on CPython 3.7, due to an interpreter bug, you will always get None".

(Actually this makes me wonder, does it work with PyPy?)

If you're willing to give me a few days I'll write a PR which attempts to implement this always-pass-none case, to see if we like it?

Reply:
Okay. Would we need the globals in that case though? I think not? so I'll wait with further changes until you do it.
If you want you can give me some pointers on how to get it done. I thought maybe setting the `Ty::Object` to be `Ty::ObjectOrInvalid`?